### PR TITLE
added Q1902

### LINKED ISSUE #3371: Add solution to lc Q1902
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Description

solution to lc Q1902

### Suggested Change

there is node code for Q1902

### Rationale

_No response_

### Urgency

Medium

### Acknowledgements

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository
- [X] I have read and followed the Contribution Guidelines
- [X] I have followed the code style guidelines of this project
- [X] I have checked for any existing open issues that my pull request may address
- [X] I have ensured that my changes do not break any existing functionality
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently
- [X] I have read the resources for guidance listed below
- [X] I have not used AI-generated content (e.g., ChatGPT, other AI tools)
- [X] I have not used content from existing sites (e.g., text, images)
- [X] I have followed all security rules and only shared trusted resources

### PR DESCRIPTION
**Pull request by Aditijainnn**
## Related Issue
 Fix #3371

## Description

added Sol  to Q1902

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+The first element of order is the root of the BST. All subsequent elements are inserted one by one, at the correct position to maintain the BST property. You need to find the depth of this binary search tree, which is the number of nodes along the longest path from the root node down to the farthest leaf node.
+
+### Examples
+
+**Example 1:**
+
+Input: order = [2,1,4,3]
+Output: 3
+Explanation: The binary search tree has a depth of 3 with path 2->3->4.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md:31]: Please remember the following text: 

Replace `2->3->4` with ``` `2->3->4` ```.

File: dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+-  order is a permutation of integers between `1` and `n`.
+
+---
+
+## Approach
+
+1- Initialization: We initiate the SortedDict with a mapping from 0 to
+   depth 0, infinity to depth 0, and the first element of the order array (which is the root node) to depth 1. Both 0 and infinity are sentinel values that assist in determining the depth of the subsequent nodes.
+
+    - sd = SortedDict({0: 0, inf: 0, order[0]: 1})
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md:58]: replace `sd = SortedDict({0: 0, inf: 0, order[0]: 1})` with ``` `sd = SortedDict({0: 0, inf: 0, order[0]: 1})` ```

File: dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+Input: order = [2,1,4,3]
+Output: 3
+Explanation: The binary search tree has a depth of 3 with path 2->3->4.
+
+**Example 2:**
+
+Input: order = [2,1,3,4]
+Output: 3
+Explanation: The binary search tree has a depth of 3 with path 2->3->4.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md:37]: Please remember the following text: 

Replace `2->3->4` with ``` `2->3->4` ```.

File: dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+**Example 3:**
+
+Input: order = [1,2,3,4]
+Output: 4
+Explanation: The binary search tree has a depth of 4 with path 1->2->3->4.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md:43]: Please remember the following text: 

Replace `1->2->3->4` with ``` `1->2->3->4` ```.

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md (hunk 4 above)
Comment (Aditijainnn) [reply dsa-solutions/lc-solutions/1900-1999/1902-depth-of-bst-given-insertion-order.md]: made all the changes

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**